### PR TITLE
Introducing clang formatting in the sdk

### LINKED ISSUE #120: Add clang-format to travis
Also add a simple `.clang-format` the teams agrees upon.

### PR DESCRIPTION
**Pull request by disa6302**
*Issue #, if available:*
#120 
*Description of changes:*
- Added clang format spec
- Included a script to clang format a file, all files in project, a directory and git diff output only
- Included a script to run clang format check locally.

To-do:
Later, https://github.com/marketplace/actions/clang-format-check can be used to enforce the check for every commit 

Resolves #120 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by MushMal** [comment]
It's hard to see what this evaluates to :). I assume this runs clean on the existing codebase?

**Comment by zhuker** [comment]
just my two cents
clang format is needed
+1 to this PR

**Comment by disa6302** [comment]
> It's hard to see what this evaluates to :). I assume this runs clean on the existing codebase?

This does not run clean because a few things are not consistent across the codebase. I have added the formatted files as well. A pretty huge PR because I ran the format on all the files.

**Comment by disa6302** [comment]
> just my two cents
> clang format is needed
> +1 to this PR

@zhuker , you are right. It is needed. Sorry this went stale for a while.

**Comment by disa6302** [comment]
> @disa6302 I think we should include the format check in this PR. Otherwise, it would be less meaningful since we might forget to run it before submitting a PR.

@lherman-cs , I will add the travis-ci portion in another PR once we confirm that the options are acceptable.

**Comment by zhuker** [comment]
> This does not run clean because a few things are not consistent across the codebase. I have added the formatted files as well. A pretty huge PR because I ran the format on all the files.

@disa6302 @MushMal 

Yeah i tried formatting and got the same - style is inconsistent across the codebase
I would modify clang format to accommodate for the most common case, meaning - `.clangformat` which produces the smallest formatting PR should win

**Comment by disa6302** [comment]
> > This does not run clean because a few things are not consistent across the codebase. I have added the formatted files as well. A pretty huge PR because I ran the format on all the files.
> 
> @disa6302 @MushMal
> 
> Yeah i tried formatting and got the same - style is inconsistent across the codebase
> I would modify clang format to accommodate for the most common case, meaning - `.clangformat` which produces the smallest formatting PR should win

Agreed. The clang format in this PR leads to the minimal changes I can see.

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+AllowShortLoopsOnASingleLine: false
+AlwaysBreakAfterDefinitionReturnType: None
+AlwaysBreakAfterReturnType: None
+AlwaysBreakBeforeMultilineStrings: false
+BinPackArguments: true
+BinPackParameters: true
+BraceWrapping:
+  AfterCaseLabel:  false
+  AfterClass:      false
+  AfterControlStatement: false
```
Comment (zhuker) [inline .clang-format:25]: my CLion complains that it should be `  AfterControlStatement: "false"` (string not bool)

Comment (disa6302) [reply .clang-format:25]: Interesting. Let me look into it.

File: samples/discoverNatBehavior.c [hunk 2]
```diff
@@ -1,36 +1,38 @@
...
             interfaceName = argv[++i];
             i++;
-        } else if (STRCMP(argv[i], STUN_HOSTNAME_PARAM) == 0) {
+        }
+        else if (STRCMP(argv[i], STUN_HOSTNAME_PARAM) == 0) {
             stunHostname = argv[++i];
             i++;
-        } else {
+        }
+        else {
```
Comment (zhuker) [inline samples/discoverNatBehavior.c:35]: if you set `BreakBeforeBraces: WebKit` in .clang-format this `else` will not be formatted

File: .clang-format [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  AfterFunction:   false
+  AfterStruct:     false
+  AfterUnion:      false
+  AfterExternBlock: false
+  BeforeCatch:     false
+  BeforeElse:      false
+  IndentBraces:    false
+  SplitEmptyFunction: false
+BreakBeforeBinaryOperators: None
+BreakBeforeBraces: Stroustrup
```
Comment (zhuker) [inline .clang-format:36]: less formatting on existing code base if BreakBeforeBraces: WebKit

File: .clang-format (hunk 1 above)
Comment (disa6302) [reply .clang-format:25]: @zhuker , 

I do not see any problem on my CLion.

File: .clang-format (hunk 3 above)
Comment (disa6302) [reply .clang-format]: Done! Thanks!

File: samples/Samples.h [hunk 4]
```diff
@@ -6,31 +6,31 @@ Shared include file for the samples
...
 #include <com/amazonaws/kinesis/video/webrtcclient/Include.h>
 
-#define NUMBER_OF_H264_FRAME_FILES                                              403
-#define NUMBER_OF_OPUS_FRAME_FILES                                              618
-#define DEFAULT_FPS_VALUE                                                       25
-#define DEFAULT_MAX_CONCURRENT_STREAMING_SESSION                                10
+#define NUMBER_OF_H264_FRAME_FILES               403
+#define NUMBER_OF_OPUS_FRAME_FILES               618
+#define DEFAULT_FPS_VALUE                        25
+#define DEFAULT_MAX_CONCURRENT_STREAMING_SESSION 10
```
Comment (zhuker) [inline samples/Samples.h:18]: `AlignConsecutiveAssignments: true` may prevent this formatting

Comment (disa6302) [reply samples/Samples.h:18]: It will. But I am going to let it stay.

File: .travis.yml [hunk 5]
```diff
@@ -176,6 +176,20 @@ matrix:
         - .github/build_windows.bat
         - cd build/tst && ./webrtc_client_test.exe --gtest_filter="-DataChannelFunctionalityTest.*:IceApiTest.*:IceFunctionalityTest.*:PeerConnectionFunctionalityTest.*:SignalingApiFunctionalityTest.*:TurnConnectionFunctionalityTest.*:RtpFunctionalityTest.marshallUnmarshallH264Data:RtpFunctionalityTest.packingUnpackingVerifySameH264Frame"
 
+    - name: clang-format Check
```
Comment (lherman-cs) [inline .travis.yml:179]: why are there 2 of this?

File: clang-format.sh [hunk 6]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
```
Comment (lherman-cs) [inline clang-format.sh:1]: I think I'm missing something. But, why do we need `check-clang.sh` when we have this script? If `check-clang.sh` is meant to be for checking `clang-format` existence, would it be better to put the script here?

Comment (disa6302) [reply clang-format.sh:1]: I am thinking the same. I am going to see which works out better. So testing it out.

**Pull request merged**